### DISCUSSION TITLE
Creating app for both Desktop and Web

### DISCUSSION THREAD

**Discussion by serhiynovos** [alex8088/electron-vite#84] [Q&A]
Is the any possibility and examples how to create both desktop and web application. Currently I have Vue app in production which can be deployed as web application or bundled with electron

**Comment by serhiynovos** [comment]
Also I see that there are no examples for Unit testings

**Reply by alex8088** [reply]
Currently, no integrated with Unit testings. It is recommended to use `playwight + Vitest`.

**Comment by alex8088** [comment]
The build `output` is like:

```
├──out/
│  ├──main
│  ├──preload
│  └──renderer
└──...
```

the `out/renderer` is the web application you want. Note that you need to make sure they are compatible in `Electron` and `browser`. 
For example, use `userAgent` to determine whether to use Electron's APIs or Web APIs.

In addition, if you are familiar with Vite, you can also add configuration file to build renderer source code. This is also possible way.
